sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
+                b = base.func(Identity(base.shape[0]))
+                if isinstance(b, MatrixBase):
```

Comment:
This would be the case if `b` is a MatrixBase. So the logic would be better as

```
if isinstance(base, MatrixBase):
    return base.func(Identity(base.shape[0]))
```

to avoid unnecessarily creating `b`. It should also remove the need to check against `TypeError`, which I'm not a fan of.